main/src/main/scala/sbt/internal/CommandExchange.scala

```diff
@@ -376,7 +377,7 @@ private[sbt] final class CommandExchange {
           .withChannelName(currentExec.flatMap(_.source.map(_.channelName)))
       case _ => pe
     }
-    if (channels.isEmpty) addConsoleChannel()
+    if (channels.isEmpty && !Terminal.isCI) addConsoleChannel()
```

Comment:
Don't we need consoles to output log messages even while we're on CI?

Reply:
In general the use of `isCI` makes me a bit nervous because it could become blind spot for testing.

Reply:
Naively you would think that but having a console channel or not does not affect whether or not System.out is available which is what is necessary for output logs. That being said, I'm not 100% sure this change is necessary so I'll see if I get hangs if I revert it.

Reply:
No hangs on my personal travis ci after reversion.